tools/checksum_bitflip/checksum_bitflip.c

```diff
@@ -0,0 +1,109 @@
+/*-
+ * Public Domain 2024-present MongoDB, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <test_util.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+/*
+ * This program takes a file and a checksum and checks whether the checksum
+ * matches the contents of the file, or if any checksum produced by flipping
+ * a single bit of the file contents matches.
+ *
+ * This is intended for use when debugging WT checksum mismatches when we suspect
+ * that the issue is the result of faulty hardware causing bit flips in memory.
+ *
+ * Usage:
+ *     checksum_bitflip cksum filename
+ *
+ * The checksum argument should be a hex string. The file should contain the data
+ * to be tested. Typical usage is to generate this data from the block dump printed
+ * when WiredTiger reports a checksum mismatch, for example by copying the hex
+ * strings into a file and running "xxd -p -r" on it.
```

Comment:
Good point. I've updated the other scripts. I've also narrowed the scope to only run in the `tools/checkpoint_bitflip` directory, rather than the entire `tools/` directory. 

It's a fair question whether we should run these scripts across the rest of the tools, but if we did, there are a bunch of strings and other things that would need to be cleaned up, and that doesn't belong in this ticket. I'll create a separate ticket and where we can decide whether and how much we want this checking on the other tools.